docs/user-guide/templates.md

```diff
@@ -0,0 +1,64 @@
+# Templates
+
+Templates are snippets of predefined code that people can use to replace a job definition in a [screwdriver.yaml](./configuration/index.md).
```

Comment:
Wasn't sure if we were going to have a full workflow template or something in the future
I can remove the extra layer if that's more clear